tests/test_python.py

```diff
@@ -109,21 +110,42 @@ def test_tox(auto_container):
     auto_container.connection.run_expect([0], "pip install --user tox")
 
 
-def test_python_devel(auto_container):
-    """Check that python_devel package can be installed."""
-    version = auto_container.connection.run_expect(
+def test_cryptography(auto_container_per_test):
+    """Check that cryptography python module can be installed from source so that
+    it is built against the SLE BCI provided openssl rather than an upstream copy."""
+    version = auto_container_per_test.connection.run_expect(
         [0], "echo $PYTHON_VERSION"
     ).stdout.strip()
-    if "3.6" in version:
-        pkg = "python3-devel"
-    elif "3.9" in version:
-        pkg = "python39-devel"
-    elif "3.10" in version:
-        pkg = "python310-devel"
-    else:
-        raise Exception(f"Unknown python version: {version}")
-    auto_container.connection.run_expect(
-        [0], f"zypper --non-interactive in {pkg}"
+
+    if packaging.version.Version(version) < packaging.version.Version("3.8"):
+        pytest.skip("cryptography tests only supported on >= 3.8")
+
+    # install dependencies
+    auto_container_per_test.connection.run_expect(
+        [0],
+        f"zypper --non-interactive in cargo libffi-devel openssl-devel gcc tar gzip",
+    )
+
+    # install latest cryptography from source
+    # TODO we should be installing a fixed version so that the 'install dependencies' steps
+    # above are working instead
```

Comment:
the reason is given in the comment "so that the install dependencies steps above are always working"

Reply:
Sorry, but I really don't see the connection here: If we pin the version, then eventually the dependencies will be bumped and will no longer work for cryptography. Also, that is going against what our users will probably do: `pip install cryptography`.

Unless you have an actionable idea here, I'd like to remove this TODO, because I really don't see how or even why it should be addressed.

Reply:
it is really unlikely that we will, in SLE15 timeframe, bump dependencies like libffi, gcc or the like to break with cryptography. it might happen with rust as it is less mature overall, but thats exactly why it should be pinned, so that we can bump them together. 

I do not believe that the majority of SLE or SLE-BCI users expect rolling release. it is more likely that they expect some version of cryptography that works with the rest of the system (like the openssl version) well. cryptography has pretty rigid dependency on the minimum openssl version, and insetalling just the latest version is gonna break completely at some point in time. 

Also, the testing is picking a subset of the tests, and hardcoding locations of the tests. these tests locations will eventually change over time, hence the suggestion to pin the version.

Reply:
Just to iterate on the last point, cryptography 39 is breaking with SLE15. so we need to pin.

Reply:
But doesn't that render this whole test moot? If we have to pin cryptography to a fixed version to make it build, then we can be pretty certain that it will break in production, because users will just try to install whatever their dependencies mandate (which eventually will be the latest version).

Reply:
Remember, the original purpose of this test is to see if `pip install --no-binary :all:` works in general (aka if python-devel and dependencies are available). cryptography is just one usecase. and yeah, if you use cryptography from source, then you're most likely to pin it because it constantly ref's its requirements, and SLE is not a rolling release distribution, so it would only work accidentally. 

Now I can install some other popular python module from source instead, or we can pin it. my preference is pinning, but that wasn't passing review, so I'll search for some other python module instead and update this.

Reply:
> the original purpose of this test is to see if `pip install --no-binary :all:` works in general

That is not really what this PR's description or the test's function name would indicate. If your use case is to just test that compilation works, then I'm fine with pinning. Please just adjust the test name, docs & commit messages accordingly.

Reply:
Okay, renamed the test function to `test_pip_install_source_cryptography` and restored the version cap again. description of the function was imho already up to date.